client/components/search/index.jsx

```diff
@@ -345,36 +325,35 @@ class Search extends Component {
 				<Spinner />
 				<div
 					className="search__icon-navigation"
-					ref="openIcon"
+					ref={ openIcon => ( this.openIcon = openIcon ) } // eslint-disable-line react/jsx-no-bind
 					onClick={ enableOpenIcon ? this.openSearch : this.focus }
 					tabIndex={ enableOpenIcon ? '0' : null }
 					onKeyDown={ enableOpenIcon ? this.openListener : null }
-					aria-controls={ 'search-component-' + this.state.instanceId }
+					aria-controls={ 'search-component-' + this.instanceId }
 					aria-label={ i18n.translate( 'Open Search', { context: 'button label' } ) }
 				>
 					{ ! this.props.hideOpenIcon && <Gridicon icon="search" className="search__open-icon" /> }
 				</div>
 				<div className={ fadeDivClass }>
 					<input
 						type="search"
-						id={ 'search-component-' + this.state.instanceId }
+						id={ 'search-component-' + this.instanceId }
+						autoFocus={ this.props.autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
+						aria-describedBy={ this.props.describedBy }
```

Comment:
Looks like there is a typo here: `aria-describedby`.

Reply:
that's just _React_ beign silly. It get's rendered correctly:
<img width="550" alt="screen shot 2018-03-16 at 14 41 28" src="https://user-images.githubusercontent.com/233601/37536111-74097aea-2928-11e8-9a53-36289d6b0a48.png">

Reply:
In fact, you are right. From the [docs](https://reactjs.org/docs/accessibility.html):

> Whereas most DOM properties and attributes in React are camelCased, these attributes should be lowercased.

Weird thing is that _React_ does not complain. Updated on 312f6d02debfb5a94ae139f710b855e2280eaf1f.